content/en/apps/guides/hosting/monitoring/postgres-ingest.md

```diff
@@ -0,0 +1,437 @@
+
+## Steps to add a query
+
+All steps are done on the CHT Watchdog instance and assume you have a credentials for your Postgres server for Watchdog to user. As well, it assumes that you installed Watchdog in `~/cht-watchdog`: 
+
+1. [Prepare query in config file]({{< relref "#prepare-query-in-config-file" >}})
+2. [Add new Postgres Exporter]({{< relref "#add-new-postgres-exporter" >}})
+3. [Adding new scrape config]({{< relref "#adding-new-scrape-config" >}})
+4. [Load new Compose files with existing ones]({{< relref "#load-new-compose-files-with-existing-ones" >}})
+5. [Configure the dashboard]({{< relref "#configure-the-dashboard" >}})
+6. [Save the dashboard for easy deployment and future updates]({{< relref "#save-the-dashboard-for-easy-deployment-and-future-updates" >}})
+
+### Prepare query in config file
+
+Add a YAML file for with your query called `~/extra-sql-queries.yml`. In this example we'll be using a query from the [App Monitoring Data Ingestion repo](https://github.com/medic/cht-app-monitoring-data-ingest/), but it can be any query as long as the user you're using in the next step has access to the database and table:
+
+```yaml
+replication_failure_reasons:
+  query: |
+    SELECT 
+      metric as failure_type,
+      sum(count) AS "count"
+    FROM 
+      public.app_monitoring_replication_failure_reasons
+    WHERE 
+      partner_name IN ('partner_name_here')
+    GROUP BY 
+      failure_type
+    ORDER BY 
+      count DESC
+  metrics:
+    - failure:
+        usage: "LABEL"
+        description: "Name of the failure"
+    - sequence:
+        usage: "TABLE"
+        description: "Replication failure reasons"
+```
+
+Note that you can use any value on the first line, seen as  `replication_failure_reasons` above.  It will be the field name we'll use in the 4th step below when exploring the new data in Grafana.
+
+### Add new Postgres Exporter
+
+In a new file  `~/extra-sql-compose.yml`, define your new Postgres exporter as well as add a mount to the existing Grafana and Prometheus services. Note that the `DATA_SOURCE_NAME` value will need to have the following variables added to you `.env` file you created to deploy Watchdog:
+
+* `EXTRA_SQL_USER` - Postgres user to use when logging in
+* `EXTRA_SQL_PASS` - Password for `EXTRA_SQL_USER` above
+* `EXTRA_SQL_SERVER` - URL or IP for your Postgres server
+* `EXTRA_SQL_PORT` - Port of server, defaults to `5432` it not declared.
+* `EXTRA_SQL_DATABASE` - Actual string of database name (eg `extra_monitoring` or `health_stats`), will be different for each install.
+
+```yaml
+services:
+
+  prometheus:
+    volumes:
+      - ./extra-sql-prometheus.yml:/etc/prometheus/scrape_configs/extra-sql-prometheus.yml:ro
+
+  grafana:
+    volumes:
+      - ./extra-sql-dashboard.json:/etc/grafana/provisioning/dashboards/CHT/cht_admin_extra_sql.json:ro
```

Comment:
```suggestion
      - ./dashboard.custom-sql.json:/etc/grafana/provisioning/dashboards/CUSTOM/custom-sql.json:ro
```

I think we should put the dashboard in a separate directory than `CHT`. The idea is that everything (alerts, dashboards, etc) in the `CHT` directory come from Watchdog. Partners wanting to add additional custom dashboards, etc should put these in different directories.

Reply:
Upon further thought, we might need to ignore this comment at least for now.  Adding a new directory to Grafana will require adding a new [`provider` instance](https://github.com/medic/cht-watchdog/blob/main/grafana/provisioning/dashboards/cht.yml).  I _think_ we could do that as needed by mapping in another `yml` config file to `/etc/grafana/provisioning/dashboards`, but I have not tried yet...

Reply:
ah - gotcha!  I'll leave as is, but just update file names per you suggestion - done in e90f21fa